File: services/academic-service/src/test/java/com/visor/school/academic/model/EnrollmentEntryTest.java
package com.visor.school.academic.model;

import org.junit.jupiter.api.Test;

import java.time.LocalDate;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

class EnrollmentEntryTest {

    @Test
    void shouldCreateEnrollmentEntryWithRequiredFields() {
        EnrollmentEntry entry = new EnrollmentEntry(
                "2024-2025",
                Term.FIRST_TERM,
                9,
                LocalDate.of(2024, 9, 1),
                EnrollmentStatus.ENROLLED
        );

        assertEquals("2024-2025", entry.getAcademicYear());
        assertEquals(Term.FIRST_TERM, entry.getTerm());
        assertEquals(9, entry.getGradeLevel());
        assertEquals(LocalDate.of(2024, 9, 1), entry.getEnrollmentDate());
        assertEquals(EnrollmentStatus.ENROLLED, entry.getStatus());
    }

    @Test
    void shouldAcceptAllTermValues() {
        List<Term> terms = List.of(
                Term.FIRST_TERM,
                Term.SECOND_TERM,
                Term.THIRD_TERM,
                Term.FULL_YEAR
        );

        for (Term term : terms) {
            EnrollmentEntry entry = new EnrollmentEntry(
                    "2024-2025",
                    term,
                    9,
                    LocalDate.of(2024, 9, 1),
                    EnrollmentStatus.ENROLLED
            );

            assertEquals(term, entry.getTerm());
        }
    }

    @Test
    void shouldAcceptAllEnrollmentStatusValues() {
        List<EnrollmentStatus> statuses = List.of(
                EnrollmentStatus.ENROLLED,
                EnrollmentStatus.GRADUATED,
                EnrollmentStatus.TRANSFERRED,
                EnrollmentStatus.WITHDRAWN
        );

        for (EnrollmentStatus status : statuses) {
            EnrollmentEntry entry = new EnrollmentEntry(
                    "2024-2025",
                    Term.FIRST_TERM,
                    9,
                    LocalDate.of(2024, 9, 1),
                    status
            );

            assertEquals(status, entry.getStatus());
        }
    }

    @Test
    void shouldAcceptGradeLevels1To12() {
        for (int grade = 1; grade <= 12; grade++) {
            EnrollmentEntry entry = new EnrollmentEntry(
                    "2024-2025",
                    Term.FIRST_TERM,
                    grade,
                    LocalDate.of(2024, 9, 1),
                    EnrollmentStatus.ENROLLED
            );

            assertEquals(grade, entry.getGradeLevel());
        }
    }
}

File: services/academic-service/src/test/kotlin/com/visor/school/academicservice/model/EnrollmentEntryTest.kt
package com.visor.school.academicservice.model

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import java.time.LocalDate

class EnrollmentEntryTest {

    @Test
    fun `should create enrollment entry with required fields`() {
        val entry = EnrollmentEntry(
            academicYear = "2024-2025",
            term = Term.FIRST_TERM,
            gradeLevel = 9,
            enrollmentDate = LocalDate.of(2024, 9, 1),
            status = EnrollmentStatus.ENROLLED
        )

        assertEquals("2024-2025", entry.academicYear)
        assertEquals(Term.FIRST_TERM, entry.term)
        assertEquals(9, entry.gradeLevel)
        assertEquals(LocalDate.of(2024, 9, 1), entry.enrollmentDate)
        assertEquals(EnrollmentStatus.ENROLLED, entry.status)
    }

    @Test
    fun `should accept all term values`() {
        val terms = listOf(
            Term.FIRST_TERM,
            Term.SECOND_TERM,
            Term.THIRD_TERM,
            Term.FULL_YEAR
        )

        terms.forEach { term ->
            val entry = EnrollmentEntry(
                academicYear = "2024-2025",
                term = term,
                gradeLevel = 9,
                enrollmentDate = LocalDate.of(2024, 9, 1),
                status = EnrollmentStatus.ENROLLED
            )

            assertEquals(term, entry.term)
        }
    }

    @Test
    fun `should accept all enrollment status values`() {
        val statuses = listOf(
            EnrollmentStatus.ENROLLED,
            EnrollmentStatus.GRADUATED,
            EnrollmentStatus.TRANSFERRED,
            EnrollmentStatus.WITHDRAWN
        )

        statuses.forEach { status ->
            val entry = EnrollmentEntry(
                academicYear = "2024-2025",
                term = Term.FIRST_TERM,
                gradeLevel = 9,
                enrollmentDate = LocalDate.of(2024, 9, 1),
                status = status
            )

            assertEquals(status, entry.status)
        }
    }

    @Test
    fun `should accept grade levels 1-12`() {
        for (grade in 1..12) {
            val entry = EnrollmentEntry(
                academicYear = "2024-2025",
                term = Term.FIRST_TERM,
                gradeLevel = grade,
                enrollmentDate = LocalDate.of(2024, 9, 1),
                status = EnrollmentStatus.ENROLLED
            )

            assertEquals(grade, entry.gradeLevel)
        }
    }
}

